examples/golang/waku.go

```diff
@@ -0,0 +1,329 @@
+	fmt.Println("Event received:", C.GoStringN(msg, C.int(len)))
+}
+
+func WakuSetEventCallback(ctx unsafe.Pointer) {
+	var str = C.allocMyString()
+	// Notice that we are not releasing the `str` memory in this case
+	// because we leave it available to the upcoming events
+	C.cGoWakuSetEventCallback(ctx, str)
+}
+
+func WakuContentTopic(ctx unsafe.Pointer,
+					  appName string,
+					  appVersion int,
+					  contentTopicName string,
+					  encoding string) string {
+	var cAppName = C.CString(appName)
+	var cContentTopicName = C.CString(contentTopicName)
+	var cEncoding = C.CString(encoding)
+	var str = C.allocMyString()
+
+	defer C.free(unsafe.Pointer(cAppName))
+	defer C.free(unsafe.Pointer(cContentTopicName))
+	defer C.free(unsafe.Pointer(cEncoding))
+	defer C.freeMyString(str)
+
+	C.cGoWakuContentTopic(ctx, 
+						cAppName,
+						C.int(appVersion),
+						cContentTopicName,
+						cEncoding,
+						str)
+
+	var contentTopic = C.GoStringN(C.getMyCharPtr(str), C.int(C.getMyCharLen(str)))
+    return contentTopic
+}
+
+func WakuPubsubTopic(ctx unsafe.Pointer, topicName string) string {
+	var cTopicName = C.CString(topicName)
+	var str = C.allocMyString()
+
+	defer C.free(unsafe.Pointer(cTopicName))
+	defer C.freeMyString(str)
+
+	C.cGoWakuPubsubTopic(ctx, cTopicName, str)
+
+	var pubsubTopic = C.GoStringN(C.getMyCharPtr(str), C.int(C.getMyCharLen(str)))
+    return pubsubTopic
+}
+
+func WakuDefaultPubsubTopic(ctx unsafe.Pointer) string {
+	var str = C.allocMyString()
+	defer C.freeMyString(str)
+	C.cGoWakuDefaultPubsubTopic(ctx, str)
+	var defaultPubsubTopic = C.GoStringN(C.getMyCharPtr(str), C.int(C.getMyCharLen(str)))
+    return defaultPubsubTopic
+}
+
+func WakuRelayPublish(ctx unsafe.Pointer,
+					pubsubTopic string,
+					message string,
+					timeoutMs int) {
+	var cPubsubTopic = C.CString(pubsubTopic)
+	var msg = C.CString(message)
+
+	defer C.free(unsafe.Pointer(cPubsubTopic))
+	defer C.free(unsafe.Pointer(msg))
+
+	C.cGoWakuRelayPublish(ctx, cPubsubTopic, msg, C.int(timeoutMs))
+}
+
+func WakuRelaySubscribe(ctx unsafe.Pointer, pubsubTopic string) {
+	var cPubsubTopic = C.CString(pubsubTopic)
+	defer C.free(unsafe.Pointer(cPubsubTopic))
+	C.cGoWakuRelaySubscribe(ctx, cPubsubTopic)
+}
+
+func WakuRelayUnsubscribe(ctx unsafe.Pointer, pubsubTopic string) {
+	var cPubsubTopic = C.CString(pubsubTopic)
+	defer C.free(unsafe.Pointer(cPubsubTopic))
+	C.cGoWakuRelayUnsubscribe(ctx, cPubsubTopic)
+}
+
+func WakuConnect(ctx unsafe.Pointer, peerMultiAddr string, timeoutMs int) {
+	var cPeerMultiAddr = C.CString(peerMultiAddr)
+	defer C.free(unsafe.Pointer(cPeerMultiAddr))
+
+	C.cGoWakuConnect(ctx, cPeerMultiAddr, C.int(timeoutMs))
+}
+
+func WakuListenAddresses(ctx unsafe.Pointer) string {
+	var str = C.allocMyString()
+	defer C.freeMyString(str)
+	C.cGoWakuListenAddresses(ctx, str)
+	var listenAddresses = C.GoStringN(C.getMyCharPtr(str), C.int(C.getMyCharLen(str)))
+    return listenAddresses
+}
+
+func main() {
+	runtime.LockOSThread()
```

Comment:
Is there any thread-local state that is being used in the underlying c bindings implementation? 
Otherwise this would not be required.

Reply:
> Is there any thread-local state that is being used in the underlying c bindings implementation? Otherwise this would not be required.

Thanks for that! I got it from the previous approach. I addressed it in https://github.com/waku-org/nwaku/pull/2559/commits/c9319c59c0d84b45e366b7567b77a28267518c0b